### PR TITLE
Ability to disable pages on prod

### LINKED ISSUE #194: Disable some features for deployment on pledge4future.org
For deployment on pledge4future.org we need to disable some features for now:
- [x] Log in
- [x] Emission planner 

@codingfabi, you had an idea of how to do this in the frontend

### PR DESCRIPTION
**Pull request by sami1riaz**
Followed this guide to implement this: https://blog.dennisokeeffe.com/blog/2021-07-25-dev-only-nextjs-routes

Prevents directly accessing the page by editing URL as well

### CODE REVIEW

File: frontend/pages/404.tsx [hunk 1]
```diff
@@ -16,6 +16,15 @@ const useStyles = makeStyles((theme) => ({
...
 }));
 
+export const getServerSideProps = () => {
+  return {
+    redirect: {
+      permanent: false,
+      destination: "/",
+    },
+  };
+};
```
Comment (sami1riaz) [inline frontend/pages/404.tsx:26]: Was initially thinking of redirecting to 404 page which is why this is implemented here. Maybe I should move it to app.tsx

File: frontend/src/api/apollo-client.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ const headerLink = setContext((_, { headers }) => {
 });
 
 const httpLink = new HttpLink({
-  uri: JSON.parse(process.env.NEXT_PUBLIC_DEV) ? `http://localhost:8000/graphql/` : `https://api.test-pledge4future.heigit.org/graphql/`
```
Comment (sami1riaz) [inline frontend/src/api/apollo-client.ts]: This is not required anymore I believe? Correct me if I'm wrong though

File: frontend/src/utils/disableOnProd.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export const disableOnProd = () => {
+  if (process.env.NODE_ENV === "production") {
```
Comment (sami1riaz) [inline frontend/src/utils/disableOnProd.ts:2]: You can test this feature locally by changing production to development

File: frontend/src/utils/disableOnProd.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    return {
+      redirect: {
+        permanent: false,
+        destination: "/",
```
Comment (sami1riaz) [inline frontend/src/utils/disableOnProd.ts]: Would it be better if I redirected to 404 instead?

File: frontend/src/api/apollo-client.ts (hunk 2 above)
Comment (codingfabi) [reply frontend/src/api/apollo-client.ts]: No we still need this during development to test against a local backend.

File: frontend/src/utils/disableOnProd.ts (hunk 4 above)
Comment (codingfabi) [reply frontend/src/utils/disableOnProd.ts]: Yes I would assume it to be quite confusing to be redirected to the main page.

**Review by codingfabi — APPROVED** [review]

**Pull request merged**